channel/state/state.go

```diff
@@ -105,11 +105,12 @@ func (s State) ChannelId() (types.Bytes32, error) {
 }
 
 func (a appPart) hash() (types.Bytes32, error) {
+
 	encodedAppPart, error := abi.Arguments{
 		{Type: uint256},
 		{Type: address},
 		{Type: bytesTy},
-	}.Pack(a.ChallengeDuration, a.AppDefinition, a.AppData)
+	}.Pack(a.ChallengeDuration, a.AppDefinition, []byte(a.AppData))
```

Comment:
@geoknee can you explain WHY this works?

Reply:
I'm not sure if I fully understand your question -- but I recall that the `.Pack` method is very picky about having exactly the right type fed into it. Here,  `a.AppData` is a custom type `types.Bytes` defined in the `types` package of our `go-nitro` module. Although the _underlying_ type is still `[]byte`, we have to explicitly convert the type to `[]byte` in order for `.Pack()` to work.

Reply:
https://github.com/statechannels/go-nitro/pull/130/commits/94e39a929839288adec7b891e51ed7e019fabe41